src/features/visual-querying/visualQuerying.slice.ts

```diff
@@ -0,0 +1,61 @@
+import type { PayloadAction } from '@reduxjs/toolkit';
+import { createSlice } from '@reduxjs/toolkit';
+
+import type { RootState } from '@/features/common/store';
+
+export enum ConstraintType {
```

Comment:
can we use a union type here, i.e. `type Constraint = 'Date' | 'Place'`?

reason is that typescript enums have a runtime cost, e.g. the above compiles to:

```js
var ConstraintType;
(function (ConstraintType) {
    ConstraintType["Date"] = "Date";
    ConstraintType["Place"] = "Place";
})(ConstraintType || (ConstraintType = {}));
```

see: https://www.typescriptlang.org/play?#code/KYOwrgtgBAwg9iAzgFwE4EMCWJkBUCeADsFAN4BQUVUAIusiQLxQDkdDLANJdQAoA26AMZNWA4cC7kAvkA